framework/web/CsvResponseFormatter.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\web;
+
+use yii\base\Arrayable;
+use yii\base\Component;
+use yii\base\InvalidConfigException;
+use yii\helpers\ArrayHelper;
+
+/**
+ * CsvResponseFormatter formats the given data into CSV response content.
+ *
+ * It is used by [[Response]] to format response data.
+ *
+ * @author Sam Mousa <sam@mousa.nl>
+ * @since 2.0.9
+ */
+class CsvResponseFormatter extends Component implements ResponseFormatterInterface
+{
+    /**
+     * @var int Maximum number of bytes to use in memory before using a temp file. Defaults to 20MB
+     */
+    public $maxMemory = 20 * 1024 * 1024;
+    /**
+     * @var string the Content-Type header for the response
+     */
+    public $contentType = 'text/csv';
+
+    /**
+     * @var boolean Whether to include column names as the first line, if data is associative.
+     */
+    public $includeColumnNames = true;
+
+    /**
+     * @var string The delimiter to use (one character only)
+     * @see fputcsv
+     */
+    public $delimiter = ',';
+
+    /**
+     * @var string The field enclosure to use (one character only)
+     * @see fputcsv
+     */
+    public $enclosure = '"';
+
+    /**
+     * @var string The escape character to use (one character only)
+     * @see fputcsv
+     */
+    public $escape = '\'';
+
+    /**
+     * @var bool Whether to check all rows for column names. This means iterating the data twice but it adds support
+     * for non-uniform data (ie rows with missing columns).
+     */
+    public $checkAllRows = false;
+
+    /**
+     * @var string The value to use for NULL values.
+     */
+    public $nullValue = "(null)";
```

Comment:
`fputcsv()` does not do anything by default, PHPs type juggling does it by default.
If you look at default behavior for gridview widgets, you see that it is customary to use a placeholder value instead of an empty string when a value is `null`.

I guess this is a matter of personal preference, @samdark if you pick I'll implement.